Array processing

Maksim-Yeager opened the pull request:
This pull request adds Javadoc comments to the methods in the ArrayProcessing class. Each method is now documented with descriptions of its functionality, parameters, return values, and any exceptions that may be thrown.

Maksim-Yeager committed:
Update .gitignore

Maksim-Yeager committed:
Update .gitignore

Maksim-Yeager committed:
Array processing is already done

I wrote the test and the methods which are explained in the desriction of the exercise

Maksim-Yeager committed:
Update .gitignore

ivorusev commented on introducing-java/src/main/java/com/onto/javacourse/intro/arrayProcessing/ArrayProcessing.java:
```diff
@@ -0,0 +1,108 @@
+/**
+ * This class provides various methods for processing arrays.
+ */
+
```
Redundant empty line.

ivorusev commented on introducing-java/src/main/java/com/onto/javacourse/intro/arrayProcessing/ArrayProcessing.java:
```diff
@@ -0,0 +1,108 @@
+        }
+
+        int minIndex = 0;
+        int minValue = array[0];
```
Do you need to store the value?

ivorusev commented on introducing-java/src/main/java/com/onto/javacourse/intro/arrayProcessing/ArrayProcessing.java:
```diff
@@ -0,0 +1,108 @@
+     * @return The sum of elements in the array.
+     * @throws IllegalArgumentException if the input array is null.
+     */
+    public static int sum(int[] array) {
```
Check this, no need to change it :) https://www.baeldung.com/java-stream-sum

ivorusev commented on introducing-java/src/main/java/com/onto/javacourse/intro/arrayProcessing/ArrayProcessing.java:
```diff
@@ -0,0 +1,108 @@
+                }
+            }
+
+            int[] subarray = Arrays.copyOfRange(array, start, end + 1);
```
It's all good. but we need the sum not the array. But I am fine with this implementation as well :)

ivorusev commented on introducing-java/src/main/java/com/onto/javacourse/intro/arrayProcessing/ArrayProcessing.java:
```diff
@@ -0,0 +1,108 @@
+
+            for (int i = 0; i < array.length; i++) {
+                currentSum += array[i];
+
```
I'd remove that empty line

ivorusev commented on introducing-java/src/main/java/com/onto/javacourse/intro/arrayProcessing/ArrayProcessing.java:
```diff
@@ -0,0 +1,108 @@
+                    start = temp;
+                    end = i;
+                }
+
```
I'd remove that empty line

ivorusev commented on introducing-java/src/test/java/com/onto/javacourse/intro/arrayProcessing/ArrayProcessingTest.java:
```diff
@@ -0,0 +1,68 @@
+        assertTrue(actualMessageBothZero.contains(expectedMessage));
+    }
+
+    // Fix it
```
:D fix it before you submit for review :D :D

ivorusev commented on introducing-java/src/main/java/com/onto/javacourse/intro/arrayProcessing/ArrayProcessing.java:
```diff
@@ -0,0 +1,108 @@
+ */
+
+public class ArrayProcessing {
+    /**
```
empty line

Maksim-Yeager committed:
code fixed completely

I removed all empty lines which were told to be and also i don't store minValue already.

Maksim-Yeager committed:
Update ArrayProcessing.java
